### PR TITLE
Add styling to decl constraints

### PR DESCRIPTION
**Pull request by landaire**
This change reduces opacity of constraints so that more emphasis is put on the function signature when reading documentation. Screenshots of the change are attached.

This change depends on https://github.com/D-Programming-Language/dmd/pull/5311.

The `:hover` isn't necessary but I thought it might be a nice touch.

hover
normal

### CODE REVIEW

**Comment by JakobOvrum** [comment]
Please also add the new `DDOC_` macro to `ddoc.dd` to document it. I think it would be fine to use the same PR for that in this case.

**Comment by JackStouffer** [comment]
I like this change, but I'm colorblind, and I have to strain my eyes to read the constraints. Could you make it so that when you hover over the signature, the opacity goes back to 100%.

**Comment by landaire** [comment]
Done and done.

**Comment by landaire** [comment]
Styling changes in the last commit make the "Constraint: " text invisible when the user's *not* hovering so that the width/text positioning does not change when the user hovers constraints

screen shot 2015-12-15 at 23 58 18
screen shot 2015-12-15 at 23 58 11

**Comment by acehreli** [comment]
Great change! Have you experimented with making the template parameter list opaque as well?

**Comment by landaire** [comment]
I haven't yet. I'm brand new to this codebase and the constraints were in a pretty obvious spot/easy to find. I'll take another look when I get home tonight (few hours) and see how easy it is.

**Comment by landaire** [comment]
@acehreli I had a message written out saying where I think you'd need to go to that but that I thought it may be outside of my scope right now. I took that as a challenge and I'm happy to say that the template parameter list is now opaque as well!

screen shot 2015-12-16 at 22 15 21

**Comment by JakobOvrum** [comment]
Template parameters often have to be explicitly provided. I don't think it's a good idea to make them opaque.

**Comment by landaire** [comment]
I'll leave that out for now, but will keep the ddoc macro that's going to be pushed to https://github.com/D-Programming-Language/dmd/pull/5311 soon

**Comment by acehreli** [comment]
Thank you landaire. Yeah, it looks weird. I think that most of the time just the regular parameters of the function are sufficient. I hope something like the following would work but it is very unusual:

    auto commonPrefix(R1 r1, R2 r2)
                     ^
       _____________/ \_____________
      (alias pred = "a == b", R1, R2)

        if (isForwardRange!R1 &&
            isInputRange!R2 &&
            !isNarrowString!R1 &&
            is(typeof(binaryFun!pred(r1.front, r2.front))));

**Comment by landaire** [comment]
Hmmm I'd need to look at that more to get a better idea of it (that kind of looks like a javascript tooltip if that was what you were going for). I'm going to stop working on this for the night but last sample: a slight color change to template parameters isn't too terrible. Maybe a different color, but you get the idea.

screen shot 2015-12-16 at 23 07 54

**Comment by JakobOvrum** [comment]
> a slight color change to template parameters isn't too terrible. Maybe a different color, but you get the idea.

Yeah, I think that's a good idea.

**Comment by acehreli** [comment]
I like it! :)

File: spec/ddoc.dd [hunk 1]
```diff
@@ -734,46 +734,48 @@ DDOC = $(LT)html$(GT)$(LT)head$(GT)
...
-DDOC_PARAM_ID         = $(DOLLAR)(TD $(DOLLAR)0)
-DDOC_PARAM_DESC       = $(DOLLAR)(TD $(DOLLAR)0)
-DDOC_BLANKLINE        = $(DOLLAR)(BR)$(DOLLAR)(BR)
-
-DDOC_ANCHOR           = $(LT)a name="$(DOLLAR)1"$(GT)$(LT)/a$(GT)
-DDOC_PSYMBOL          = $(DOLLAR)(U $(DOLLAR)0)
-DDOC_PSUPER_SYMBOL    = $(DOLLAR)(U $(DOLLAR)0)
-DDOC_KEYWORD          = $(DOLLAR)(B $(DOLLAR)0)
-DDOC_PARAM            = $(DOLLAR)(I $(DOLLAR)0)
+DDOC_COMMENT            = $(LT)!-- $(DOLLAR)0 --$(GT)
```
Comment (WalterBright) [inline spec/ddoc.dd]: What's different about these lines?

Comment (JackStouffer) [inline spec/ddoc.dd]: Whitespace, see https://github.com/landaire/dlang.org/commit/8c5fd3505afffacd42214d1f6e3048b8d73a3461#diff-3e7607c8dfde1f2374db45a77861cd3eL737

Comment (landaire) [inline spec/ddoc.dd]: Here's the same diff with whitespace ignored (note the `?w=1`) https://github.com/D-Programming-Language/dlang.org/pull/1169/files?w=1#r48080375

**Comment by landaire** [comment]
Here's the final change to styling: 

screen shot 2015-12-21 at 21 45 14

Template parameter lists also have a title of "Template parameter list" when you hover.

If this is good with you guys then this should be good when https://github.com/D-Programming-Language/dmd/pull/5311 is merged.

**Comment by JackStouffer** [comment]
Please squish all commits and get rid of the whitespace changes I pointed out, then LGTM.

**Comment by adamdruppe** [comment]
I want to reiterate that I think it is awfully silly that we're worried about maintaining backward compatibility with the ugly, poorly marked up, barely usable default doc output.

**Comment by landaire** [comment]
@JackStouffer done and done.

**Comment by JackStouffer** [comment]
LGTM

**Comment by landaire** [comment]
@JackStouffer I made a quick change per the feedback in the last-pass code review in dmd. [Link to the diff](https://github.com/landaire/dlang.org/commit/0d89ca9ec07c53581597fdb2b28b58eba8e6c3f8) (I already did a squash).

**Comment by landaire** [comment]
https://github.com/D-Programming-Language/dmd/pull/5311 was merged so this should be GTG whenever.

**Comment by JackStouffer** [comment]
This looks great, especially with the font change.

LGMT

**Comment by Hackerpilot** [comment]
:white_check_mark:

**Comment by JackStouffer** [comment]
@CyberShadow @MartinNowak ping. Can this be merged before 2.070?

File: css/style.css [hunk 2]
```diff
@@ -364,6 +364,29 @@ dd, .d_decl_dd
     margin-bottom: 1em;
 }
 
+.template_param_list
```
Comment (CyberShadow) [inline css/style.css]: Nit, inconsistent brace style

**Pull request merged**